fullscreen/api/historical.html

```diff
@@ -35,4 +35,8 @@
     // it is convenient to just test all names on both <video> and document.
   });
 });
+
+test(function() {
+  assert_false('fullScreen' in window);
+}, 'window.fullScreen must not be supported');
```

Comment:
I don't quite understand why we need this test here.

`window.fullScreen` in Gecko is **not** a historical implementation of the Fullscreen API. It was dated back way earlier than Fullscreen API, and actually was there before Firefox. It was initially implemented in a [bug](https://bugzilla.mozilla.org/show_bug.cgi?id=68136) 20 years ago to allow the browser window to go full screen, and this API probably has been leaked to the web content since then.

So I don't think this test makes sense here, as it's unrelated to the Fullscreen API.

Reply:
Do you mean it doesn't return the same value as `document.fullscreen`? I think https://github.com/mozilla/gecko-dev/blob/2ec947a6853ee7fca5997ff5e45bd45b6afd6218/dom/base/nsGlobalWindowInner.cpp#L3655-L3664 is the implementation but I can't figure out where it ends up really returning a value.

If it's truly uncoupled to the Fullscreen API, then I agree a negative test doesn't belong here.

Reply:
It does not return the same value as `document.fullscreen`. So basically, when `document.fullscreen` returns `true`, `window.fullScreen` returns `true` as well, because the browser window is also full screen, but not vice versa. If you use F11 to make the browser window full screen on a website which doesn't use Fullscreen API, then `window.fullScreen` also returns `true` while `document.fullscreen` would be `false`.

In that sense, they are not completely uncoupled, but they are different things.

Reply:
Hmm, I wonder why this difference isn't resulting in failing tests. Or even worse, maybe the WebDriver tests depend on the Firefox behavior for window.fullScreen...

Needs investigation!

Reply:
Maybe just because nothing tried to make the browser enter full screen mode without using the Fullscreen API.

Reply:
It looks like the tests are for https://w3c.github.io/webdriver/#dfn-fullscreen-window, which is defined in terms of https://fullscreen.spec.whatwg.org/#fullscreen-an-element, fortunately it's not exercising the browser UI fullscreen feature which isn't observable to web content.